feat: make topics multilevel and manageable through CMS

arslanashraf7 opened the pull request:
#### Pre-Flight checklist

- [ ] Screenshots and design review for any changes that affect layout or styling
  - [ ] Desktop screenshots
  - [ ] Mobile width screenshots
- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested
- [ ] Settings
  - [ ] New settings are documented and present in `app.json`
  - [ ] New settings have reasonable development defaults, if applicable

#### What are the relevant tickets?
#2593

#### What's this PR do?
- Makes topics model to support multi-level topics
- Adds topics management through CMS
- Allow courseware CMS pages to allow associating topics to a course

#### How should this be manually tested?
(Required)

#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
(Optional)

#### Screenshots (if appropriate)
(Optional)

#### What GIF best describes this PR or how it makes you feel?
(Optional)

asadali145 commented on courses/models.py:
```diff
@@ -290,6 +290,13 @@ class CourseTopic(TimestampedModel):
     name = models.CharField(max_length=128, unique=True)
+    parent = models.ForeignKey(
+        "self",
+        on_delete=models.CASCADE,
```
Could we add a more useful message for the user while deleting topics? Maybe we should mention that it might delete child courses.
<img width="886" alt="Screen Shot 2023-03-28 at 4 48 34 PM" src="https://user-images.githubusercontent.com/52656433/228226671-060bb8b6-6554-4eaf-b2ee-e8c295ff144f.png">

arslanashraf7 replied:
Done in https://github.com/mitodl/mitxpro/pull/2596/commits/ceb557cfb593b48ddb98e4a6d4188fa7f4b644c8

asadali145 reviewed (approved):
Overall LGTM! Just a couple of suggestions.

arslanashraf7 commented:
@asadali145 Thanks for the review, I've addressed the feedback in https://github.com/mitodl/mitxpro/pull/2596/commits/ceb557cfb593b48ddb98e4a6d4188fa7f4b644c8 and also update a screenshot in PR description.

Could you take a look again?

asadali145 reviewed (approved):
LGTM! great work!
